### PR TITLE
Allow initialization of a `GeoDataFrame` using a `cudf.DataFrame`

### LINKED ISSUE #893: [FEA]: Create `GeoDataFrame` from `cudf.DataFrame`
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Critical (currently preventing usage)

### Please provide a clear description of problem you would like to solve.

It is desirable that we can create a `GeoDataFrame` from a `cudf.DataFrame`. One can perform certain cudf operations like `.merge` on a `GeoDataFrame`, which at present returns a `DataFrame`, which doesn't understand how to operate on the `geometry` column that it contains. At the least as a stopgap measure we should be able to use `cuspatial.GeoDataFrame(cudf_dataframe)` to create a new wrapper around a `cudf.DataFrame` so that cuspatial methods will work on it.

### Describe any alternatives you have considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by thomcom**
## Description
Certain cudf operations like `.merge` return a `cudf.DataFrame`, which then doesn't understand the `geometry` column contained within. Until we've worked out a way to build `ExtensionDtypes` for cudf, the `cudf.DataFrame` needs to be converted back to a `GeoDataFrame` for use with cuspatial. This PR adds a new valid constructor argument of type `cudf.DataFrame` to `__init__` for `GeoDataFrame`.

Closes #893 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/cuspatial/core/_column/geocolumn.py [hunk 1]
```diff
@@ -179,6 +179,20 @@ def copy(self, deep=True):
...
+        """
+        return self._meta.input_types.valid_count
+
+    def has_nulls(self) -> bool:
+        """
+        Returns True if any of the geometries stored in this GeoColumn are
+        null.
+        """
+        return self._meta.input_types.has_nulls
+
```
Comment (isVoid) [inline python/cuspatial/cuspatial/core/_column/geocolumn.py:195]: Let's make sure these are not included in this PR. They belong to #894

**Review by isVoid — APPROVED** [review]
There are code that should be included in the other PR that's subject to removal.

File: python/cuspatial/cuspatial/core/_column/geocolumn.py (hunk 1 above)
Comment (thomcom) [reply python/cuspatial/cuspatial/core/_column/geocolumn.py:195]: :+1:

**Review by harrism — APPROVED** [review]

File: python/cuspatial/cuspatial/core/_column/geocolumn.py (hunk 1 above)
Comment (thomcom) [inline python/cuspatial/cuspatial/core/_column/geocolumn.py:195]:
```suggestion
    @property
    def valid_count(self) -> int:
        """
        Returns the number of valid geometries stored in this GeoColumn.
        """
        return self._meta.input_types.valid_count

    def has_nulls(self) -> bool:
        """
        Returns True if any of the geometries stored in this GeoColumn are
        null.
        """
        return self._meta.input_types.has_nulls
```

**Pull request merged**